### PR TITLE
Tests that network seed always affects dna hash

### LINKED ISSUE #2039: [BUG] Network seed is not changing the DNA hash when source is `AppBundleSource::Bundle(AppBundle)`
**Describe the bug**
Installing an app using `bundle` instead of `path` results in the same DNA hash even when `network_seed` is also provided.

**Expected behavior**
I expect that providing a `network_seed` will always change the DNA hash regardless of the app source used.

**System information:**
 - OS: ubuntu 22.04
 - Holochain v0.1.0 

**Steps to reproduce**
When installing an app using [`AppBundleSource::Bundle(AppBundle)`](https://github.com/holochain/holochain/blob/develop/crates/holochain_types/src/app.rs#L154) and a `network_seed` the network seed is ignored.  The DNA hash is the same as without the network seed.

#### Call 1 - bundle without network seed

```js
{
    "installed_app_id": "appstore-alice",
    "agent_key": "uhCAkU9Lo-H0vGzZNWB7116BpborSAItxYq8DDTHFaStSyEaBYxGh",
    "membrane_proofs": {},
    "network_seed": null,
    "bundle": {
        "manifest": {
            "manifest_version": "1",
            "name": "appstore",
            "roles": [
                {
                    "name": "appstore",
                    "dna": {
                        "path": "/projects/app-store-dnas/bundled/appstore.dna"
                    }
                },
                {
                    "name": "portal",
                    "dna": {
                        "path": "/projects/app-store-dnas/bundled/portal.dna"
                    }
                }
            ]
        },
        "resources": {}
    }
}
```

Portal DNA hash == `uhC0knmUIHLo7p1XORNSHReovyinmj7WmRGef82nyoUvzDt2LJZRT` (correct)

#### Call 2 - bundle with network seed

```js
{
    "installed_app_id": "appstore-alice",
    "agent_key": "uhCAknz5iK5FFMKn_v9RTmfo3wgrDWKipUUiRkG3UH-t9GhWKaUP7",
    "membrane_proofs": {},
    "network_seed": "test-network",
    "bundle": {
        "manifest": {
            "manifest_version": "1",
            "name": "appstore",
            "roles": [
                {   
                    "name": "appstore",
                    "dna": {
                        "path": "/projects/app-store-dnas/bundled/appstore.dna"
                    }
                },
                {   
                    "name": "portal",
                    "dna": {
                        "path": "/projects/app-store-dnas/bundled/portal.dna"
                    }
                }
            ]
        },
        "resources": {}
    }
}
```

Portal DNA hash == `uhC0knmUIHLo7p1XORNSHReovyinmj7WmRGef82nyoUvzDt2LJZRT` (incorrect)

#### Call 3 - path with network seed

```js
{
    "installed_app_id": "appstore-alice",
    "agent_key": "uhCAkjDQY6E7heEv8iF2f7lB-tYnfQ5iGvdxyc39xomAAlLTrFQx7",
    "membrane_proofs": {},
    "network_seed": "test-network",
    "path": "/projects/app-store-dnas/appstore.happ"
}
```

Portal DNA hash == `uhC0kWQfOGS-ruIu-mOu_BC4yns-xhcCrahinLpPpVuunOGqrLLBL` (correct)

The hash from call 2 should be the same as call 3.

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Fixes #2039 

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_types/src/app/app_manifest/app_manifest_v1.rs [hunk 1]
```diff
@@ -195,7 +195,11 @@ impl AppManifestV1 {
     // we know we need it, since this way is substantially simpler.
     pub fn set_network_seed(&mut self, network_seed: NetworkSeed) {
         for mut role in self.roles.iter_mut() {
-            if matches!(role.provisioning, Some(CellProvisioning::Create { .. })) {
+            if !matches!(
+                role.provisioning.clone().unwrap_or_default(),
+                CellProvisioning::CreateClone { .. } | CellProvisioning::UseExisting { .. }
+            ) {
```
Comment (maackle) [inline crates/holochain_types/src/app/app_manifest/app_manifest_v1.rs:201]: This is the fix

File: crates/holochain/src/conductor/tests/install_app_bundle.rs [hunk 2]
```diff
@@ -129,3 +132,339 @@ async fn reject_duplicate_app_for_same_agent() {
...
+                agent_key,
+                source,
+                installed_app_id: Some(case_str.clone()),
+                network_seed,
+                membrane_proofs: HashMap::new(),
+            })
+            .await
+            .unwrap();
+
+        println!("finished case {}", case_str);
```
Comment (jost-s) [inline crates/holochain/src/conductor/tests/install_app_bundle.rs:465]:
```suggestion
```

**Review by jost-s — APPROVED** [review]

File: crates/holochain_types/src/combinators.rs [hunk 3]
```diff
@@ -24,3 +24,8 @@ pub fn second_ref<A, B>(tup: &(A, B)) -> &B {
 pub fn swap2<A, B>(tup: (A, B)) -> (B, A) {
     (tup.1, tup.0)
 }
+
+/// Map an iterator into a vec of a new type
+pub fn mapvec<'a, T: 'a, U>(it: impl Iterator<Item = &'a T>, f: impl FnMut(&'a T) -> U) -> Vec<U> {
```
Comment (ThetaSinner) [inline crates/holochain_types/src/combinators.rs:29]: Should this be `map_vec`?

File: crates/holochain/src/conductor/tests/install_app_bundle.rs [hunk 4]
```diff
@@ -129,3 +132,339 @@ async fn reject_duplicate_app_for_same_agent() {
...
+    // if both of these apps can be installed under the same agent, the
+    // network seed change was successful -- otherwise there will be a
+    // CellAlreadyInstalled error.
+
+    let _app1 = conductor
+        .clone()
+        .install_app_bundle(InstallAppPayload {
+            agent_key: agent.clone(),
+            source: AppBundleSource::Bundle(bundle1),
+            installed_app_id: Some("no-seed".into()),
```
[the remark this answers is not in the document]
Comment (maackle) [reply crates/holochain/src/conductor/tests/install_app_bundle.rs:192]: In the more comprehensive ignored test, I used "no seed", "seed A", and "seed B". But in this case, it really is about seed vs no seed, and in this case there is really no seed.

File: crates/holochain_types/src/combinators.rs (hunk 3 above)
Comment (maackle) [reply crates/holochain_types/src/combinators.rs:29]: It is what it is :)

**Pull request merged**